### PR TITLE
(FACT-2072) Update the regex to get Infiniband MAC and IP address on CentOS 7.x platform

### PR DESCRIPTION
**Pull request by raj76**
This pr updates the regular expression used to find and replace the mac address as shown by the ifconfig command.The ifconfig output changed on CentOS 7 platforms and the old regular expression did not work anymore. I've changed the conditional logic to look for ib* instead of just ib.

### CODE REVIEW

File: lib/facter/util/ip.rb [hunk 1]
```diff
@@ -161,9 +161,9 @@ def self.get_single_interface_output(interface)
       output = Facter::Util::IP.ifconfig_interface(interface)
     when 'Linux'
       ifconfig_output = Facter::Util::IP.ifconfig_interface(interface)
-      if interface =~ /^ib/ then
+      if interface =~ /^ib.*/ then
```
Comment (ciprianbadescu) [inline lib/facter/util/ip.rb]: /^ib/ is equivalent to /^ib.*/ as used here, why is this change needed?

Comment (raj76) [reply lib/facter/util/ip.rb]: It was a change I made when trying to figure out the problem. Since they are equivalent this line can be reverted. Thanks!

Comment (GabrielNagy) [reply lib/facter/util/ip.rb]: You can revert this line since it makes no difference

Comment (raj76) [reply lib/facter/util/ip.rb]: I've reverted this change.
